staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -213,24 +242,27 @@ func (s *SpdyRoundTripper) proxyAuth(proxyURL *url.URL) string {
 // clients may call SpdyRoundTripper.Connection() to retrieve the upgraded
 // connection.
 func (s *SpdyRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	// TODO what's the best way to clone the request?
-	r := *req
-	req = &r
-	req.Header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
-	req.Header.Add(httpstream.HeaderUpgrade, HeaderSpdy31)
+	header := utilnet.CloneHeader(req.Header)
+	header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
+	header.Add(httpstream.HeaderUpgrade, HeaderSpdy31)
 
-	conn, err := s.dial(req)
+	conn, rawResponse, err := utilnet.ConnectWithRedirects(req.Method, req.URL, header, req.Body, s)
```

Comment:
Yeah. It was beta in 1.6, it seems reasonable to remove it in 1.7. But the flag was guarding the idea of the apiserver sniffing the responses for redirects. Since that's what this file is doing now, it seems like it should be guarded by the same feature flag (as long as the feature flag exists).